### PR TITLE
New filter utility class for xss

### PR DESCRIPTION
**Pull request by Drullan**
Added a utility class for filtering. Added method that filters dangerous xss-tags. Filtered entries with the new method.

### CODE REVIEW

File: src/main/java/se/lth/cs/connect/routes/Entry.java [hunk 1]
```diff
@@ -317,10 +318,10 @@ protected void setup(PippoSettings conf) {
             update.add(DO.SET(entry.property("contact")).to(e.contact));
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/routes/Entry.java:318]: Not sure if jcypher will do its own checking here (e.contact might be malicious), maybe worth checking out

File: src/main/java/se/lth/cs/connect/routes/Entry.java [hunk 2]
```diff
@@ -159,16 +160,16 @@ public IClause build(JcNode node) {
...
-                    .property("doi").value(doi)
+                    .property("reference").value(Filter.xss(reference))
+                    .property("doi").value(Filter.xss(doi))
                     .property("hash").value(hval)
                     .property("date").value(date);
             else
                 return CREATE.node(node).label("entry").label(entryType)
                     .property("contact").value(contact)
                     .property("hash").value(hval)
                     .property("date").value(date)
```
Comment (smetz) [inline src/main/java/se/lth/cs/connect/routes/Entry.java]: We should also filter `date`, `entryType` and `contact`.

File: src/main/java/se/lth/cs/connect/routes/Entry.java [hunk 3]
```diff
@@ -159,16 +160,16 @@ public IClause build(JcNode node) {
...
-                    .property("description").value(description);
+                    .property("description").value(Filter.xss(description));
```
Comment (smeets) [inline src/main/java/se/lth/cs/connect/routes/Entry.java:172]: We should also filter `date`, `contact` and `entryType`

**Comment by smeets** [comment]
[#44 New filter utility class for xss](https://trello.com/c/894uYrXZ/83-44-new-filter-utility-class-for-xss)

**Pull request closed without merging**